src/RestSharp/RestClient.cs

```diff
@@ -128,7 +128,7 @@ public partial class RestClient : IDisposable {
     void ConfigureHttpClient(HttpClient httpClient) {
         if (Options.MaxTimeout > 0) httpClient.Timeout = TimeSpan.FromMilliseconds(Options.MaxTimeout);
 
-        if (Options.UserAgent != null && httpClient.DefaultRequestHeaders.UserAgent.All(x => x.Product?.Name != Options.UserAgent)) {
+        if (Options.UserAgent != null && httpClient.DefaultRequestHeaders.UserAgent.All(x => $"{x.Product?.Name}/{x.Product?.Version}" != Options.UserAgent)) {
```

Comment:
Good catch :) How about moving the default user agent to a get-only private property, so we have this calculated in one place? The code which sets the header and checks if it's already there will use the same value even if we decide to change how it's composed.

Reply:
I'm not sure I fully understand your idea, and I believe it's because I'm tripping myself up mentally due to the fact that the default user agent is already a `static readonly` field:

https://github.com/restsharp/RestSharp/blob/f7d3984244c4808ae7c068e028753c2857f325b9/src/RestSharp/RestClientOptions.cs#L26-L28

Are you proposing that we encapsulate the user agent so that the composition and equality checks are handled from one place? I can certainly implement such a feature, I just want to make sure I'm comprehending your request before I get started.

Can you elaborate for clarity?

Reply:
Yeah, I haven't looked. Even better. I mean replacing
```
$"{x.Product?.Name}/{x.Product?.Version}"
```
with
```
DefaultUserAgent
```

Reply:
A few questions:

1. Are you comfortable with the encapsulated approach?
2. Should we handle this in `v109` as well?
3. If yes to both, I must create a separate pull request for `v109` since the `RestSharpOptions` file is stored in that branch's `Options` folder. I recommend that we create the encapsulation at the root where `RestSharpOptions` can be found in `dev` so at least that part merges smoothly, but the implementation in `RestSharpOptions` would need to be implemented by hand in both.

I'm ready to start working on this locally.

Reply:
It's fine to merge to main as it won't introduce conflicts with v109. There's no need to create a separate PR for v109 branch.

Please ignore the whole conversation here, I didn't realise what your code is doing and I get it now. I think it's great, please change nothing. It's all correct. Sorry about the confusion.

Reply:
```csharp
using System.Net.Http.Headers;

namespace RestSharp;

/// <summary>
/// Represents a user agent for RestSharp.
/// </summary>
public sealed class RestSharpUserAgent
{
    private static readonly string _version = typeof(RestClientOptions).Assembly.GetName()?.Version?.ToString() ?? "0.0.0.0";
    /// <summary>
    /// Initializes a new instance of the <see cref="RestSharpUserAgent"/> class.
    /// </summary>
    /// <param name="name">The name of the user agent.</param>
    /// <param name="version">The version of the user agent.</param>
    public RestSharpUserAgent(string name, string version) {
        Name = name;
        Version = version;
    }
    /// <summary>
    /// The name of the user agent.
    /// </summary>
    public string Name { get; }
    /// <summary>
    /// The version of the user agent.
    /// </summary>
    public string Version { get; }
    /// <summary>
    /// The default <see cref="RestSharpUserAgent"/>.
    /// </summary>
    /// <returns></returns>
    public static RestSharpUserAgent Default => new RestSharpUserAgent("RestSharp", _version);
    /// <summary>
    /// Returns a <see cref="string"/> representing the name and version of the <see cref="RestSharpUserAgent"/>.
    /// </summary>
    /// <remarks>This will return <see cref="Name"/>/<see cref="Version"/>.</remarks>
    /// <returns>A <see cref="string"/> representing the name and version of the <see cref="RestSharpUserAgent"/>.</returns>
    public override string ToString() => $"{Name}/{Version}";
    /// <summary>
    /// Determines whether the specified <see cref="object"/> is equal to the current <see cref="RestSharpUserAgent"/>.
    /// </summary>
    /// <param name="obj">The <see cref="object"/> to compare the current <see cref="RestSharpUserAgent"/> with.</param>
    /// <returns><see langword="true"/> if the <see cref="object"/> is equal to the current <see cref="RestSharpUserAgent"/>, otherwise <see langword="false"/>.</returns>
    public override bool Equals(object obj)
    {
        if (obj is string objString)
            return objString.Equals(ToString());

        if (obj is RestSharpUserAgent agent)
            return agent.Name.Equals(Name) && agent.Version.Equals(Version);

        if (obj is ProductInfoHeaderValue productHeader)
            return productHeader.Product.Name.Equals(Name) && productHeader.Product.Version.Equals(Version);

        return obj.Equals(this);
    }
    public static implicit operator string(RestSharpUserAgent agent) => agent.ToString();
    public static implicit operator RestSharpUserAgent(string agent)
    {
        if (agent is null || string.IsNullOrWhiteSpace(agent) || !agent.Contains('/'))
            return Default;

        IEnumerable<string> split = agent.Split('/');
        if (split.Count() != 2)
            return Default;

        return new RestSharpUserAgent(split.First(), split.Last());
    }
}
```

There's no discussion area, so I can't post what I had already there. Here's the start of an encapsulated approach, in-case the project ever goes in that direction.

Reply:
You can just use a record to avoid explicit equality implementation.